handlers/archives.go

```diff
@@ -0,0 +1,171 @@
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const archiveAPIURL = "https://web.archive.org/cdx/search/cdx"
+
+func convertTimestampToDate(timestamp string) (time.Time, error) {
+	year, err := strconv.Atoi(timestamp[0:4])
+	if err != nil {
+		return time.Time{}, err
+	}
+	month, err := strconv.Atoi(timestamp[4:6])
+	if err != nil {
+		return time.Time{}, err
+	}
+	day, err := strconv.Atoi(timestamp[6:8])
+	if err != nil {
+		return time.Time{}, err
+	}
+	hour, err := strconv.Atoi(timestamp[8:10])
+	if err != nil {
+		return time.Time{}, err
+	}
+	minute, err := strconv.Atoi(timestamp[10:12])
+	if err != nil {
+		return time.Time{}, err
+	}
+	second, err := strconv.Atoi(timestamp[12:14])
+	if err != nil {
+		return time.Time{}, err
+	}
+	return time.Date(year, time.Month(month), day, hour, minute, second, 0, time.UTC), nil
+}
```

Comment:
It's a small thing, but I'd probably put the minuteIndex, hourIndex, dayIndex, etc as variables, to avoid magic numbers. Because, for example `hour, err := strconv.Atoi(timestamp[8:10])` is a bit hard to read.

```go
const (
	yearIndex          = 0
	monthIndex         = 4
	dayIndex           = 6
	hourIndex          = 8
	minuteIndex        = 10
	secondIndex        = 12
	timestampLength    = 14
	averagePageSizeDiv = 100
)
```